Allow Hermes to check RabbitMQ status

cdonati opened the pull request:
This adds a `/stats/cluster/rabbitmq` endpoint that returns a structure like
```
{"nodes": [{"mem_alarm": false,
            "name": "rabbit@appscale-image0",
            "disk_free_alarm": false}],
 "queues": [{"messages": 0,
             "name": "appscaledashboard___default"}]}
```
If either mem_alarm or disk_free_alarm is `true` for any of the nodes, the TQ servers will be unable to enqueue messages.

I'm not attached to the endpoint. It just happens that `/stats/cluster...` is accessible from the AdminServer port already.

jeanleonov commented on Hermes/appscale/hermes/stats/producers/rabbitmq.py:
```diff
@@ -0,0 +1,78 @@
+    headers = {'Authorization': 'Basic {}'.format(creds)}
+    try:
+      response = yield self.client.fetch(url, headers=headers)
+    except Exception as error:
```
Does tornado http client raise error if it receives 400+ code?

cdonati commented:
@jeanleonov I moved things around to more closely match the other producers. Let me know if that works. I removed the queue stats for now since they aren't divisible by machine. I'll proabably submit a separate pull request with a partial implementation of https://cloud.google.com/appengine/docs/standard/python/taskqueue/rest/taskqueues/get for push queues.

jeanleonov commented on Hermes/appscale/hermes/stats/producers/rabbitmq_stats.py:
```diff
@@ -0,0 +1,87 @@
+class RabbitMQStatsSnapshot(object):
+  """ A container for holding RabbitMQ node stats. """
+  utc_timestamp = attr.ib()
+  rabbitmq_node_stats = attr.ib(metadata={Meta.ENTITY_LIST: RabbitMQNodeStats})
```
@cdonati , I don't think that nested list is required here. Do we expected that there can be multiple node stats objects? Anyway, `RabbitMQStatsSource.get_current` method puts single object into this attribute.
So I would say that `RabbitMQNodeStats` and `RabbitMQStatsSnapshot` can be merged into single class with node stats and timestamp.

jeanleonov commented on Hermes/appscale/hermes/stats/producers/rabbitmq_stats.py:
```diff
@@ -0,0 +1,77 @@
+                          '{}: {}'.format(url, response.body))
+
+    snapshot = RabbitMQStatsSnapshot(
+      utc_timestamp=int(time.time()),
```
oh.. this was so simple.. and I was using `utc_timestamp = time.mktime(datetime.now().timetuple())`. (facepalm).

cdonati commented:
https://ocd.appscale.com:8080/job/Daily%20Build/4558/
